server/integration/utils/auth.ts

```diff
@@ -0,0 +1,74 @@
+import passportOAuth1 from 'passport-oauth1';
+
+import { User, Integration, IntegrationDataOAuth1 } from 'server/models';
+import { isDuqDuq, isDevelopment } from 'utils/environment';
+
+const baseRedirectUrl = isDuqDuq()
+	? 'https://duqduq.org'
+	: isDevelopment()
+	? 'http://lvh.me:9876'
+	: 'https://pubpub.org';
+
+export const zoteroAuthStrategy = () =>
+	new passportOAuth1.Strategy(
+		{
+			requestTokenURL: 'https://www.zotero.org/oauth/request',
+			accessTokenURL: 'https://www.zotero.org/oauth/access',
+			userAuthorizationURL: 'https://www.zotero.org/oauth/authorize',
+			consumerKey: process.env.ZOTERO_CLIENT_KEY,
+			consumerSecret: process.env.ZOTERO_CLIENT_SECRET,
+			callbackURL: `${baseRedirectUrl}/auth/zotero/redirect`,
+			signatureMethod: 'HMAC-SHA1',
+			passReqToCallback: true,
+		},
+		(req, token, tokenSecret, params, profile, cb) => {
+			const userId = req.user.id;
+			const { username: externalUsername, userID: externalUserId } = params;
+			const externalUserData = {
+				externalUserId,
+				externalUsername,
+			};
+			const integrationDataOAuth1 = {
+				userId,
+				accessToken: token,
+			};
+			return User.findOne({
+				where: { id: userId },
+				include: {
+					model: Integration,
+					where: { name: 'zotero' },
+					required: false,
+					include: { model: IntegrationDataOAuth1, required: false },
+				},
+			})
+				.then((user) => {
+					if (!user.integrations.length) {
+						return user
+							.createIntegration({
+								name: 'zotero',
+								authSchemeName: 'OAuth1',
+								externalUserData,
+							})
+							.then((integration) =>
+								integration.createIntegrationDataOAuth1(integrationDataOAuth1),
+							);
+					}
+					const integration = user.integrations[0];
+					return integration
+						.getIntegrationDataOAuth1()
+						.then((oldData) => {
+							if (!oldData) {
```

Comment:
Would this ever happen? i.e. would a "zotero" integration ever be created without an `IntegrationDataOAuth1`? If not, we may want use the `assert` utility here to assert that `oldData` is always truthy. That way we're throwing an error intentionally if the integration ends up in an invalid state.